crates/ruff_linter/resources/test/fixtures/pyupgrade/UP046.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any, AnyStr, Generic, ParamSpec, TypeVar, TypeVarTuple
+
+S = TypeVar("S", str, bytes)  # constrained type variable
+T = TypeVar("T", bound=float)
+Ts = TypeVarTuple("Ts")
+P = ParamSpec("P")
+
+
+class A(Generic[T]):
+    # Comments in a class body are preserved
+    pass
+
+
+class B(Generic[*Ts]):
+    pass
+
+
+class C(Generic[P]):
+    pass
+
+
+class Constrained(Generic[S]):
+    pass
+
+
+# This case gets a diagnostic but not a fix because we can't look up the bounds
+# or constraints on the generic type from another module
+class ExternalType(Generic[T, SupportsRichComparisonT]):
+    pass
+
+
+# typing.AnyStr is a common external type variable, so treat it specially as a
+# known TypeVar
+class MyStr(Generic[AnyStr]):
+    pass
+
+
+class MultipleGenerics(Generic[S, T, Ts, P]):
+    pass
+
+
+# These cases are not handled
+class D(Generic[T, T]):  # duplicate generic variable, runtime error
+    pass
+
+
+# TODO(brent) we should also apply the fix to methods, but it will need a
+# little more work. these should be left alone for now but be fixed eventually.
+class NotGeneric:
+    # -> generic_method[T: float](t: T)
+    def generic_method(t: T):
+        pass
```

Comment:
micro-nit: it's nice to keep fixtures looking like "realistic" Python code where possible (this looks a bit strange to me, as there's generally no reason to use a `TypeVar` in a function's annotations like this unless either two parameters are annotated with the `TypeVar` or the return annotation uses the `TypeVar`):

```suggestion
    def generic_method(var: T) -> T:
        return var
```

this applies to a few other functions and classes in your fixtures, but I won't comment on all of them ;)

Reply:
Ah sorry for making you repeat yourself. I know you mentioned it for the docs, but I thought I could get away with it for the tests! I'll keep this in mind now.